Andy/complete cancel pickup events

WishingWell13 opened the pull request:
<!-- Fill in all spots surrounded by brackets and confirm all check boxes after confirming completion of the tasks -->

# Info

Closes #219 .

<!-- If there is no issue for this pull request yet, please create one or
delete this line if the pull request is for a very minor tweak. -->

Fixed Linked Event Dropdown Breaks After Click and No Select For Admin Pickup Event Page

<!-- What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context. -->

## Changes

- Made entering weird dates also not break the admin portal

# Type of Change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) <!-- Edit this type of change if you select this -->

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [ ] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots

<!-- If you made any visual changes to the website, please include relevant screenshots below. -->

(ignore timer, is an extension)
<img width="1512" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/bc25e87e-e240-4dc4-af17-45812e9f112d">

WishingWell13 committed:
Added complete and cancel buttons to PickupEvent Edit Page

Also fixed text wrap issue

Also updated Button component to take in class names

WishingWell13 committed:
Added buttons to admin pickup home page

WishingWell13 committed:
Merge branch 'main' into andy/completeCancelPickupEvents

WishingWell13 committed:
Moved complete and edit buttons to single pickup event page

Updated button styling for pickup event page

Updated common Button component styling to auto expand to content

Hide the complete/cancel buttons if the event isn't active

WishingWell13 committed:
Merge branch 'main' into andy/completeCancelPickupEvents

WishingWell13 committed:
Removed unnecessary CSS

WishingWell13 committed:
Merge branch 'andy/completeCancelPickupEvents' of https://github.com/acmucsd/membership-portal-ui-v2 into andy/completeCancelPickupEvents

WishingWell13 committed:
Merge branch 'main' into andy/completeCancelPickupEvents

WishingWell13 committed:
Fixed bug with linkedEventUuid not found edge case

If you clicked the select a pickup event toggle but didn't select an event, the API call wouldn't work b/c the linkedEventUuid would be '', not null.

Fixed the issue by changing value to be null when linkedEventUuid is ''

WishingWell13 committed:
Made admin pickup events sort by start date

WishingWell13 committed:
Removed console.log

(For last commit)
https://stackoverflow.com/questions/54634515/typescript-sorting-object-by-date

raymosun reviewed (changes requested):
Sorting looks good!

For `linkedEventUUID`, maybe instead of creating a new variable, replace each instance with `linkedEventUUID || null`.

I also noticed another bug: if you spam garbage in the date, it can break the site. Maybe 
![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/30709485/b812f42f-44b9-47d5-a63b-68e482cd6736)

![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/30709485/71d47566-3b53-44b4-84ca-f5d7683072bd)

WishingWell13 committed:
Refactored code to eliminate creation of new variable

You have to specify the key name
like linkedEventUuid: linkedEventUuid || null
the shorthand only works if you pass in a single variable that’s used as the key and value like u did for title

WishingWell13 committed:
Added checks to make weird dates not break code

added a popup instead

WishingWell13 committed:
Use instanceof instead of === for error check

alexzhang1618 commented on src/components/admin/event/AdminPickupEvent/AdminPickupEventForm.tsx:
```diff
@@ -128,13 +132,20 @@ const AdminPickupEventForm = ({ mode, defaultData = {}, token, upcomingEvents }:
     try {
+      const start = new Date(isoStart).toISOString();
+      const end = new Date(isoEnd).toISOString();
+      const orderLimit = parseInt(`${rawOrderLimit}`, 10);
```
I see you're doing this above as well, can we condense/reuse?

WishingWell13 committed:
Added function to parse props

for complete and cancel functions

raymosun reviewed (approved):
Looks good to me 🚀
